File: LeetCode Problems/1901 to 2000/1903. Largest Odd Number in String/1903. Largest Odd Number in String.java
class Solution {
    public String largestOddNumber(String num) {
        int pos = -1;
        for (int i = 0; i < num.length(); i++) {
            if ((num.charAt(i) - '0') % 2 == 1) {
                pos = i;
            }
        }
        if (pos < 0) return "";
        return num.substring(0, pos + 1);
    }
}

File: LeetCode Problems/1901 to 2000/1903. Largest Odd Number in String/1903. Largest Odd Number in String.cpp
class Solution {
    bool isOdd(int num) {
        return num % 2;
    }

public:
    string largestOddNumber(string num) {
        int pos = -1;
        for (int i = 0; i < num.size(); i++) {
            if (isOdd(num[i] - '0')) {
                pos = i;
            }
        }
        return num.substr(0, pos + 1);
    }
};